code/modules/xenoarchaeology/tools/xenoarchaeology_labeler.dm

```diff
@@ -0,0 +1,261 @@
+/obj/item/xenoarchaeology_labeler
+	name = "artifact labeler"
+	icon = 'icons/obj/xenoarchaeology/xenoartifact_tech.dmi'
+	icon_state = "labeler"
+	desc = "A tool scientists use to label their alien bombs."
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_TINY
+	///Checked traits
+	var/list/selected_traits = list()
+	///List of deselected traits
+	var/list/deselected_traits = list()
+	///List of selected traits we'll put on the label
+	var/list/label_traits = list()
+	///List of possible trait filters
+	var/list/trait_filters = list(
+	list("icon" = "eye", "desc" = "Traits that can appear in the material description."),
+	list("icon" = "hand-sparkles", "desc" = "Traits that can be detected by 'feeling' the artifact."),
+	list("icon" = "wrench", "desc" = "Traits that can be triggered with specific items."),
+	list("icon" = "search", "desc" = "Traits that can be detected with specific items."),
+	list("icon" = "clone", "desc" = "Traits that have 'clones' or 'twins'."),
+	list("icon" = "dice", "desc" = "Traits with randomized effects."),
+	list("icon" = "snowflake", "desc" = "Traits that spawn particles, or change the artifact's appearance."),
+	list("icon" = "volume-up", "desc" = "Traits that passively make noise"))
```

Comment:
```suggestion
	var/list/trait_filters = list(
		list("icon" = "eye", "desc" = "Traits that can appear in the material description."),
		list("icon" = "hand-sparkles", "desc" = "Traits that can be detected by 'feeling' the artifact."),
		list("icon" = "wrench", "desc" = "Traits that can be triggered with specific items."),
		list("icon" = "search", "desc" = "Traits that can be detected with specific items."),
		list("icon" = "clone", "desc" = "Traits that have 'clones' or 'twins'."),
		list("icon" = "dice", "desc" = "Traits with randomized effects."),
		list("icon" = "snowflake", "desc" = "Traits that spawn particles, or change the artifact's appearance."),
		list("icon" = "volume-up", "desc" = "Traits that passively make noise")
	)
```
nitpick, but an additional indent makes it better to read

Reply:
nuh uh